### PR TITLE
Add OIE Java embedded auth SDK/SIW sample app guide 3

### PR DESCRIPTION
**Pull request by vanngo-okta**
## Description:
- **What's changed?** Add Java content to OIE embedded auth widget sign-in and social sign-in use cases. Updated sign-in use case code sample based on Arvind's feedback.
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-408182](https://oktainc.atlassian.net/browse/OKTA-408182)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md [hunk 1]
```diff
@@ -42,53 +41,23 @@ mav.addObject(REDIRECT_URI,
...
-            new HttpEntity<>(requestParams, headers);
-   final String tokenUri = ClientUtil.getNormalizedUri(issuer, "/v1/token");
-
-   final ResponseEntity<JsonNode> responseEntity =
-            restTemplate.postForEntity(tokenUri, requestEntity, JsonNode.class);
-
-   return responseEntity.getBody();
-}
-```
+Use the interaction code and code verifier to request for tokens. See helper class [exchangeCodeForToken()](https://github.com/okta/okta-idx-java/blob/f9378d48d39c10c76294e079f35214bbef3a02cd/samples/embedded-sign-in-widget/src/main/java/com/okta/spring/example/HelperUtil.java#L80).
```
Comment (arvindkrishnakumar-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md:48]:
```suggestion
Use the interaction code and code verifier to request for tokens. See helper function [exchangeCodeForToken()](https://github.com/okta/okta-idx-java/blob/f9378d48d39c10c76294e079f35214bbef3a02cd/samples/embedded-sign-in-widget/src/main/java/com/okta/spring/example/HelperUtil.java#L80).
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md [hunk 2]
```diff
@@ -42,53 +41,23 @@ mav.addObject(REDIRECT_URI,
...
 
 ```java
 final JsonNode jsonNode = helperUtil.exchangeCodeForToken(interactionCode, codeVerifier);
 final OAuth2AccessToken oAuth2AccessToken = helperUtil.buildOAuth2AccessToken(jsonNode);
 final OAuth2RefreshToken oAuth2RefreshToken = helperUtil.buildOAuth2RefreshToken(jsonNode);
-
 ```
 
-### Step 4: Call user profile information (Optional)
+The Spring security framework doesn't understand Okta’s interaction code flow. Therefore, the app needs to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, fill the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) to hand over authentication flow back to Spring.
```
Comment (arvindkrishnakumar-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md:56]:
```suggestion
The Spring security framework does not understand Okta’s Interaction code flow. Therefore, your app will need to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, fill the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) before handing over the authentication flow back to Spring.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+   request.getContextPath() + "/authorization-code/callback"
+);
+mav.addObject(ISSUER_URI, issuer);
+
+session.setAttribute(CODE_VERIFIER, idxClientContext.getCodeVerifier());
+```
+
+### Step 6: Request tokens from Okta
+
+Use the interaction code and code verifier to request for tokens. See helper class [exchangeCodeForToken()](https://github.com/okta/okta-idx-java/blob/f9378d48d39c10c76294e079f35214bbef3a02cd/samples/embedded-sign-in-widget/src/main/java/com/okta/spring/example/HelperUtil.java#L80).
```
Comment (arvindkrishnakumar-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md:64]:
```suggestion
Use the interaction code and code verifier to request for tokens. See helper function [exchangeCodeForToken()](https://github.com/okta/okta-idx-java/blob/f9378d48d39c10c76294e079f35214bbef3a02cd/samples/embedded-sign-in-widget/src/main/java/com/okta/spring/example/HelperUtil.java#L80).
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+```java
+final JsonNode jsonNode = helperUtil.exchangeCodeForToken(interactionCode, codeVerifier);
+final OAuth2AccessToken oAuth2AccessToken = helperUtil.buildOAuth2AccessToken(jsonNode);
+final OAuth2RefreshToken oAuth2RefreshToken = helperUtil.buildOAuth2RefreshToken(jsonNode);
+```
+
+The Spring security framework doesn't understand Okta’s interaction code flow. Therefore, the app needs to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, fill the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) to hand over authentication flow back to Spring.
```
Comment (arvindkrishnakumar-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md:72]: Same edit suggestions as https://github.com/okta/okta-developer-docs/pull/2327/files#r669707789

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+### Step 7: Retrieve user profile
+
+Retrieve the user profile attributes with the access token object and populate Spring framework’s [OAuth2AuthenticationToken](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java)  object reference for Spring to continue with the rest of the authentication flow.
```
Comment (arvindkrishnakumar-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md:76]: Would be nice to link HelperUtil class's `getUserAttributes` method here.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md (hunk 2 above)
Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md:56]:
```suggestion
The Spring security framework doesn't understand Okta’s Interaction code flow. Therefore, your app needs to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, populate the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) before handing over the authentication flow back to Spring.
```

Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/java/integrationsteps.md:56]: Sorry Arvind, I'm following our style guide convention to use contractions and present, active voice as much as possible.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md (hunk 4 above)
Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md:72]:
```suggestion
The Spring security framework doesn't understand Okta’s Interaction code flow. Therefore, your app needs to intercept Spring’s OAuth authentication code flow, exchange the interaction code obtained from Okta for an access token, populate the user profile attributes, and construct [OAuth2AuthenticationToken.java](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java) before handing over the authentication flow back to Spring.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md (hunk 5 above)
Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md:76]:
```suggestion
Retrieve the user profile attributes with the access token object and populate Spring framework’s [OAuth2AuthenticationToken](https://github.com/spring-projects/spring-security/blob/main/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2AuthenticationToken.java)  object reference for Spring to continue with the rest of the authentication flow. See helper class method [getUserAttributes()](https://github.com/okta/okta-idx-java/blob/f9378d48d39c10c76294e079f35214bbef3a02cd/samples/embedded-sign-in-widget/src/main/java/com/okta/spring/example/HelperUtil.java#L67) for details. 
```

**Pull request merged**